### PR TITLE
Feature/api response in data obj

### PR DESCRIPTION
**Pull request by esteban-gs**
Changes
-----------
With the [swagger feature added](https://github.com/SavageLearning/Machete/pull/637), some API controllers were not returning records inside a data `JSON` object as we have historically been doing. The Angular client is already set up to look for records inside a `data` object in the API JSON response.

Example JSON response from current production version v `1.14.285`:
```
{
  "data": [
    {
      "createdby": null,
      "datecreated": "1900-01-01T00:00:00",
      "dateupdated": "1900-01-01T00:00:00",
      "id": 0,
      "idPrefix": null,
      "idString": null,
      "updatedby": null,
      "active": true,
      "category": "null",
      "emailTemplate": null,
      "fixedJob": null,
      "key": null,
      "level": null,
      "ltrCode": null,
      "minHour": null,
      "minimumCost": null,
      "selected": false,
      "skillDescriptionEn": null,
      "skillDescriptionEs": null,
      "sortorder": 0,
      "speciality": false,
      "subcategory": null,
      "text_EN": "null",
      "text_ES": "null",
      "typeOfWorkID": 0,
      "wage": null,
      "clientRules": null
    }
]
}

```
This pull request is primarily aimed at getting the API responses to wrap the returned records inside the `JSON` data object and unit test all API Rest controllers.

New:
=======
`ApiRequestParams`
----------------------
Why: Although, we currently don't use server-side pagination at the Rest API level, I added it there
Testing: to make testing easier; passing an empty object instead of many params:   https://github.com/SavageLearning/Machete/blob/432086126356d59f6ffa2ed315bf1b7bfd84d0ea/Machete.Test/UnitTests/Web/Controllers/Api/WorkAssignmentsControllerTests.cs#L83

Swagger: By having this be a ViewModel, Swagger will document it as a schema model
image
```
"ApiRequestParams": {
        "type": "object",
        "properties": {
          "pageNumber": {
            "type": "integer",
            "format": "int32"
          },
          "pageSize": {
            "type": "integer",
            "format": "int32"
          },
          "sortField": {
            "type": "string",
            "nullable": true
          },
          "sortDesc": {
            "type": "boolean"
          },
          "search": {
            "type": "string",
            "nullable": true
          },
          "dwccardnum": {
            "type": "integer",
            "format": "int32",
            "nullable": true
          },
          "memberStatusId": {
            "type": "integer",
            "format": "int32",
            "nullable": true
          }
        },
```
image

`PaginationMetadata`
--------------------------
why: to make testing API response easier. It's simpler to deserialize an API response into an object than to check if the JSON string matches the expected string. We can avoid doing this:
https://github.com/SavageLearning/Machete/blob/432086126356d59f6ffa2ed315bf1b7bfd84d0ea/Machete.Test/UnitTests/Web/Controllers/WorkOrderControllerTests.cs#L149

Swagger: It's getting added to the `swagger.json` through:
https://github.com/SavageLearning/Machete/blob/432086126356d59f6ffa2ed315bf1b7bfd84d0ea/Machete.Web/Controllers/Api/PaginationMetaDataSchemaFilter.cs#L1-L17

-All selenium tests passing-

### CODE REVIEW

File: Machete.Web/Controllers/Api/Abstracts/MacheteApi2Controller.cs [hunk 1]
```diff
@@ -15,7 +16,8 @@ namespace Machete.Web.Controllers.Api
     // 
     // Controllers that extend this class will need to `new` an override method that calls the base method for each
     // method that is mean to be re-used.
-    public abstract class MacheteApi2Controller<TD, TVM> : ControllerBase where TD : Record
+    public abstract class MacheteApi2Controller<TD, TVM> : ControllerBase 
+        where TD : Record where TVM : class
```
Comment (jcii) [inline Machete.Web/Controllers/Api/Abstracts/MacheteApi2Controller.cs:20]: pretty sure there's a companium RecordVM that you can limit TVM to...

**Review by chaim1221 — DISMISSED** [review]
sorry I didn't get back to this on Sunday...

To summarize what I said in Slack, basically the approach here is too large. If individual json objects need to be manipulated let's start there. We shouldn't be changing the tests, and the client, and this all at once, and if we do change the controllers, those changes should have a very limited scope.

I will find some time to make an example before I pick anything else up, and understand the problem a bit better.

**Review by chaim1221 — COMMENTED** [review]
so I'm removing my blocking review but this doesn't necessarily mean I have approved... I basically need a chunk of time to review this and I have been oncall etc.; if it gets to a point where @jcii is happy then I'm happy too, but I would like to review it if possible.

File: appveyor_prebuild.sh [hunk 2]
```diff
@@ -6,8 +6,8 @@ dotnet --version
 dotnet clean
 cd UI
 npm install
-npm run --silent build-local-dev
-npm run start-local-dev &
+npm run --silent build-prod
+npm run build-prod
```
Comment (esteban-gs) [inline appveyor_prebuild.sh:10]: You don't need the `&` in there?

**Pull request closed without merging**